File: Maths/continous_subarray_sum.py
class Solution(object):
    def __init__(self,nums,k:int):
        self.nums = nums
        self.k = k
    def answer(self,nums,k:int)->int:
        self.n = len(self.nums)
        for i in range(self.n):
            if(self.nums[i]%self.k==0):
                continue
            self.sum = self.nums[i]
            for j in range(i+1,self.n):
                self.sum = self.sum+self.nums[j]
                if(self.sum%self.k==0):
                    return True
        return False


if __name__=='__main__':
    arr = [23,2,6,4,7]
    k = 6
    s = Solution(arr,k)
    print(s.answer(arr,k)) # True

File: Maths/continous_subarray_sum.cxx
#include <bits/stdc++.h>
using namespace std;

class Solution{
    public:
    static bool checkSubarraySum(vector<int> &nums, int k){
        for(auto i = 0; nums[i]; ++i){
            if(nums[i]%k==0)
                continue;
            int sum = nums[i];
            for(auto j = i+1; nums[j]; ++j){
                sum = sum + nums[j];
                if(sum%k==0)
                    return true;
            }
        }
        return false;
    }
};
int main(){
    ios_base::sync_with_stdio(0);
    vector<int> nums = {23, 2, 6, 4, 7};
    int k = 6;
    cout << Solution::checkSubarraySum(nums,k);
}
